internal/querycoordv2/observers/task_dispatcher.go

```diff
@@ -0,0 +1,101 @@
+// Licensed to the LF AI & Data foundation under one
+// or more contributor license agreements. See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership. The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package observers
+
+import (
+	"context"
+	"sync"
+
+	"github.com/milvus-io/milvus/pkg/util/conc"
+	"github.com/milvus-io/milvus/pkg/util/typeutil"
+)
+
+// taskDispatcher is the utility to provide task dedup and dispatch feature
+type taskDispatcher[K comparable] struct {
+	tasks      *typeutil.ConcurrentSet[K]
+	pool       *conc.Pool[any]
+	notifyCh   chan struct{}
+	taskRunner task[K]
+	wg         sync.WaitGroup
+	cancel     context.CancelFunc
+	stopOnce   sync.Once
+}
+
+type task[K comparable] func(context.Context, K)
+
+func newTaskDispatcher[K comparable](runner task[K]) *taskDispatcher[K] {
+	return &taskDispatcher[K]{
+		tasks:      typeutil.NewConcurrentSet[K](),
+		pool:       conc.NewPool[any](8),
```

Comment:
I believe 8 is too small for this, as the observer executes IO operations mostly, let's make it larger

Reply:
no sure which value is suitable here

Reply:
max(32, CpuNum() * 2)

Reply:
changed to configurable item and increase default to 16